tests/api/gef_memory.py

```diff
@@ -0,0 +1,147 @@
+"""
+`gef.session` test module.
+"""
+
+import pathlib
+import random
+import pytest
+
+from tests.base import RemoteGefUnitTestGeneric
+
+from tests.utils import (
+    debug_target,
+    gdbserver_session,
+    qemuuser_session,
+    GDBSERVER_DEFAULT_HOST,
+)
+
+
+class GefMemoryApi(RemoteGefUnitTestGeneric):
+    """`gef.memory` test module."""
+
+    def setUp(self) -> None:
+        self._target = debug_target("default")
+        return super().setUp()
+
+    def test_api_gef_memory_only_running(self):
+        gdb, gef = self._gdb, self._gef
+
+        with pytest.raises(RuntimeError):
+            assert gef.memory.maps is None
+
+        gdb.execute("start")
+        assert gef.memory.maps is not None
+
+    def test_api_gef_memory_parse_info_proc_maps_expected_format(self):
+        if self.gdb_version < (11, 0):
+            pytest.skip(f"Skipping test for version {self.gdb_version} (min 10.0)")
+
+        gdb, root = self._gdb, self._conn.root
+        gdb.execute("start")
+
+        # Check output format
+        lines = (gdb.execute("info proc mappings", to_string=True) or "").splitlines()
+        assert len(lines) >= 5
+        assert all(map(lambda x: isinstance(x, str), lines))
+        for line in lines[4:]:
+            parts = [x.strip() for x in line.split()]
+            start_addr = int(parts[0], 16)
+            end_addr = int(parts[1], 16)
+            size = int(parts[2], 16)
+            int(parts[3], 16)
+            assert end_addr == start_addr + size
+            assert len(parts[4]) == 4, parts[4]
```

Comment:
What's with this second part of the assertion? That's the error message. We could probably give a better explanation (or drop it altogether)

Reply:
Bad debug print, fixed